File: gaia-sdk-javascript/src/graphql/response/type/Code.ts


import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {Order} from "../../request/enumeration/Order";
import {OrderByField} from "../../request/enumeration/OrderByField";
import {EdgeOrderByField} from "../../request/enumeration/EdgeOrderByField";

/**
* Represents code information
*/
export interface Code {
    /**
    * The code id
    */
    identityId?:Uuid, 
    /**
    * The code reference id
    */
    reference?:Uuid, 
    /**
    * The name of the code
    */
    qualifier?:string, 
    /**
    * Detailed description about the code
    */
    appendent?:string, 
    /**
    * The code dictionary. The key is a file name and the value is the code
    */
    code?:Struct, 
    /**
    * The list of labels of the code
    */
    labelList?:[string], 
    /**
    * The type of the code
    */
    type?:string
}

File: gaia-sdk-python/gaia_sdk/graphql/response/type/Code.py


from typing import List
Uuid = str
String = str
ISO8601 = str
Struct = dict
Float = float
Int = int
Boolean = bool
from gaia_sdk.graphql.request.enumeration.Order import Order
from gaia_sdk.graphql.request.enumeration.OrderByField import OrderByField
from gaia_sdk.graphql.request.enumeration.EdgeOrderByField import EdgeOrderByField
from gaia_sdk.graphql.request.enumeration.EdgeType import EdgeType

class Code:
    """
    Represents code information
    """
    dictionary: dict

    def __init__(self, dictionary: dict):
        self.dictionary = dictionary

    def __eq__(self, other):
        if type(other) is type(self):
            return self.dictionary == other.dictionary
        return False

    def __repr__(self):
        return {'dictionary': self.dictionary}

    """
    The code id
    """
    @property
    def identity_id(self) -> Uuid:
        return Uuid(self.dictionary.get("identityId"))
    """
    The code reference id
    """
    @property
    def reference(self) -> Uuid:
        return Uuid(self.dictionary.get("reference"))
    """
    The name of the code
    """
    @property
    def qualifier(self) -> String:
        return String(self.dictionary.get("qualifier"))
    """
    Detailed description about the code
    """
    @property
    def appendent(self) -> String:
        return String(self.dictionary.get("appendent"))
    """
    The code dictionary. The key is a file name and the value is the code
    """
    @property
    def code(self) -> Struct:
        return Struct(self.dictionary.get("code"))
    """
    The list of labels of the code
    """
    @property
    def label_list(self) -> List[String]:
        return list(map(lambda x: String(x), self.dictionary.get("labelList")))
    """
    The type of the code
    """
    @property
    def type(self) -> String:
        return String(self.dictionary.get("type"))
